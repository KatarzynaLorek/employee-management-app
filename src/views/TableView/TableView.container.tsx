import React from 'react';
import { useAppDispatch, useAppSelector } from '../../hooks/reduxHooks';
import { addEmployee, removeEmployee, updateEmployee } from '../../store/actions/actions';
import { IResponseObject } from '../../types/responses';
import TableView from './TableView';

const TableViewContainer: React.FC = () => {
  const employeesData: IResponseObject[] = useAppSelector((state) => state.rootReducer.employees);
  const dispatch = useAppDispatch();

  const handleAddEmployee = (employee: IResponseObject): Promise<any> =>
    dispatch(addEmployee(employee));

  const handleUpdateEmployee = (employee: IResponseObject): Promise<any> =>
    dispatch(updateEmployee(employee));

  const handleRemoveEmployee = (employee: IResponseObject): Promise<any> =>
    dispatch(removeEmployee(employee.id));

  return (
    <TableView
      employeesData={employeesData}
      handleAddEmployee={handleAddEmployee}
      handleUpdateEmployee={handleUpdateEmployee}
      handleRemoveEmployee={handleRemoveEmployee}
    />
  );
};

export default TableViewContainer;
