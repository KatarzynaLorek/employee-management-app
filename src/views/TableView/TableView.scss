.tableview {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 1000px;

  &__navigation {
    display: flex;
    flex-direction: row;
    align-self: flex-end;
    margin: 30px 20px 0 0;
  }

  &__button {
    margin: 0 5px;
    width: 70px;
    height: 40px;
    border-radius: 9px;
    background-color: #fff;
    border: 2px solid #77b28c;
    text-transform: uppercase;
    line-height: 40px;
    transition: 0.1s linear;
    cursor: pointer;

    &:hover {
      background-color: #77b28c;
      color: white;
    }
    &--add {
      @extend .tableview__button;
      width: 100px;
      align-self: flex-start;
      margin-bottom: 10px;
      margin-left: 50px;
    }
  }
}

@media (max-width: 1200px) {
  .tableview {
    width: 900px;
    padding-top: 100px;

    &__navigation {
      margin-right: 20px;
    }

    &__button {
      &--add {
        margin-left: 40px;
      }
    }
  }
}

@media (max-width: 1000px) {
  .tableview {
    width: 750px;

    &__navigation {
      margin-right: 0;
    }

    &__button {
      &--add {
        margin-left: 25px;
      }
    }
  }
}

@media (max-width: 768px) {
  .tableview {
    margin-top: 120px;
    width: 700px;

    &__navigation {
      margin: 15px 20px 0 0;
    }
  }
}
