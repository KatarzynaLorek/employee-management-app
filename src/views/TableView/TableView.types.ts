import { IResponseObject } from '../../types/responses';

export interface ITableView {
  tableData: IResponseObject[];
  addEmployee: (employee: IResponseObject) => void;
  updateEmployee: (employee: IResponseObject) => void;
  removeEmployee: (employee: IResponseObject) => void;
}
