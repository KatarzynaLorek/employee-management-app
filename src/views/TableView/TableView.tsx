import React from 'react';
import Table from '../../components/Table/Table';
import './TableView.scss';
import EmployeeForm from '../../components/EmployeeForm/EmployeeForm';
import { IResponseObject } from '../../types/responses';

export interface ITableView {
  isAddFormOpen: boolean;
  isUpdateFormOpen: boolean;
  updatedEmployee: IResponseObject | null;
  tableData: IResponseObject[];
  handleChangeAddFormVisibility: () => void;
  handleChangeUpdateFormVisibility: (employee?: IResponseObject) => void;
  handlePageChange: (type: 'next' | 'prev') => void;
  currentPage: number;
  maxPageNumber: number;
  handleSubmit: (newValues: IResponseObject, setSubmitting: (boolean: boolean) => void) => void;
  handleRemoveEmployee: (employee: IResponseObject) => Promise<any>;
  newID: string;
}

const TableView = ({
  isAddFormOpen,
  isUpdateFormOpen,
  updatedEmployee,
  tableData,
  handleChangeAddFormVisibility,
  handleChangeUpdateFormVisibility,
  handlePageChange,
  currentPage,
  maxPageNumber,
  handleSubmit,
  newID,
  handleRemoveEmployee,
}: ITableView): JSX.Element => {
  return (
    <div className="tableview">
      {isAddFormOpen && (
        <EmployeeForm handleClose={handleChangeAddFormVisibility} handleSubmit={handleSubmit} />
      )}
      {isUpdateFormOpen && (
        <EmployeeForm
          newID={newID}
          employeeData={updatedEmployee}
          handleSubmit={handleSubmit}
          handleClose={handleChangeUpdateFormVisibility}
        />
      )}
      <button
        className="tableview__button tableview__button--add"
        onClick={handleChangeAddFormVisibility}
      >
        Add
      </button>
      <Table
        tableData={tableData}
        handleOpenUpdateForm={handleChangeUpdateFormVisibility}
        handleRemoveEmployee={handleRemoveEmployee}
      />
      <div className="tableview__navigation">
        <button
          className="tableview__button"
          onClick={() => handlePageChange('prev')}
          disabled={currentPage === 1}
        >
          Prev
        </button>
        <button
          className="tableview__button"
          onClick={() => handlePageChange('next')}
          disabled={currentPage === maxPageNumber}
        >
          Next
        </button>
      </div>
    </div>
  );
};

export default TableView;
