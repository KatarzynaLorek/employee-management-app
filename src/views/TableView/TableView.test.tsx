import React from 'react';
import { render } from '@testing-library/react';
import TableView from './TableView';

describe('TableView component', () => {
  it('renders properly', () => {
    const { getByText } = render(<TableView />);
    expect(getByText('Add')).toBeInTheDocument();
  });
});
