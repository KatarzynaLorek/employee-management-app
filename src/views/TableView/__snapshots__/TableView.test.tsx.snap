// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableView component renders properly 1`] = `
<body>
  <div>
    <div
      class="tableview"
    >
      <button
        class="tableview__button--add"
      >
        Add
      </button>
      <div
        class="table"
      >
        <div
          class="row--title"
        >
          <div
            class="row__part"
          >
            <p
              class="row__label"
            >
              First Name
            </p>
            <p>
              First Name
            </p>
          </div>
          <div
            class="row__part"
          >
            <p
              class="row__label"
            >
              Last Name
            </p>
            <p>
              Last Name
            </p>
          </div>
          <div
            class="row__part"
          >
            <p
              class="row__label"
            >
              Department
            </p>
            <p>
              Department
            </p>
          </div>
          <div
            class="row__part"
          >
            <p
              class="row__label"
            >
              Position
            </p>
            <p>
              Position
            </p>
          </div>
          <div
            class="row__part"
          >
            Actions
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="row__part"
          >
            <p
              class="row__label"
            >
              First Name
            </p>
            <p>
              mockFirstName
            </p>
          </div>
          <div
            class="row__part"
          >
            <p
              class="row__label"
            >
              Last Name
            </p>
            <p>
              mockLastname
            </p>
          </div>
          <div
            class="row__part"
          >
            <p
              class="row__label"
            >
              Department
            </p>
            <p>
              IT
            </p>
          </div>
          <div
            class="row__part"
          >
            <p
              class="row__label"
            >
              Position
            </p>
            <p>
              junior
            </p>
          </div>
          <div
            class="row__part"
          >
            <button
              class="row__button--update"
            >
              Update
            </button>
            <button
              class="row__button--delete"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
      <div
        class="tableview__navigation"
      >
        <button
          class="tableview__button"
          disabled=""
        >
          Prev
        </button>
        <button
          class="tableview__button"
          disabled=""
        >
          Next
        </button>
      </div>
    </div>
  </div>
</body>
`;
